### PR TITLE
[CCP-136] Initial commit from CCP-136 old branch

### PR DESCRIPTION
**Pull request by mkarthik90**
All the changes from CCP-136 is moved to the new branch
CCP-136 new branch. All the changes are moved due to
rebasing and merge issues. Once CCP-136 new branch is merged
to develop. Both CCP-136 and CCP-136-New-Branch can be removed.

### CODE REVIEW

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/root/DatabaseManager.java [hunk 1]
```diff
@@ -25,6 +25,9 @@ public void init() {
 		configuration.file().blockSize(80);
 		configuration.common().objectClass(ConceptEntry.class).objectField("wordnetId").indexed(true);
 		configuration.common().objectClass(ConceptEntry.class).objectField("id").indexed(true);
+        // Added to make sure list has been added to the database
+        // Ref: http://www.resolvinghere.com/sof/12343387.shtml
+        configuration.common().objectClass(ConceptEntry.class).updateDepth(2);
```
Comment (mkarthik90) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/root/DatabaseManager.java]: This shows as a formatting change, because other lines are formatted with tab spacing.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/ChangeEventConstants.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package edu.asu.conceptpower.servlet.core;
+
+public interface ChangeEventConstants {
```
Comment (mkarthik90) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/ChangeEventConstants.java]: Should I use an enum instead of this constant file?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/root/DatabaseManager.java (hunk 1 above)
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/root/DatabaseManager.java]: indentation

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/ChangeEvent.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    @Override
+    public int compareTo(ChangeEvent o) {
+        ChangeEvent changeEvent = (ChangeEvent) o;
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/ChangeEvent.java]: why do you cast ```o``` here?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/ChangeEventConstants.java (hunk 2 above)
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/ChangeEventConstants.java]: ya, that would probably work nicely.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/ConceptManager.java [hunk 4]
```diff
@@ -381,17 +383,15 @@ public String addConceptListEntry(ConceptEntry entry)
...
-		entry.setId(id);
-		client.store(entry, DBNames.DICTIONARY_DB);
-		if (entry.getWordnetId() != null) {
-		    String wordnetId = entry.getWordnetId();
-		    if (wordnetId.endsWith(",")) {
-		        wordnetId = wordnetId.substring(0, wordnetId.length()-1);
-		    }
-		    indexService.deleteById(wordnetId);
-		}
-		indexService.insertConcept(entry);
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/ConceptManager.java]: why did you delete this?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/ConceptManager.java [hunk 5]
```diff
@@ -381,17 +383,15 @@ public String addConceptListEntry(ConceptEntry entry)
...
+        List<ChangeEvent> changeEventList = new ArrayList<ChangeEvent>();
+        changeEventList.add(changeEvent);
+        entry.setChangeEvents(changeEventList);
+        String id = generateId(CONCEPT_PREFIX);
+        entry.setId(id);
+        indexService.insertConcept(entry);
+        client.store(entry, DBNames.DICTIONARY_DB);
         return id;
 
 	}
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/ConceptManager.java]: also, formatting is off.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/IndexService.java [hunk 6]
```diff
@@ -120,4 +120,20 @@ public IndexingEvent getTotalNumberOfWordsIndexed() {
...
     }
 
+    /**
+     * This method updates the concept in lucene index based on id of the
+     * concept
+     * 
+     * @throws IllegalAccessException
+     */
+    @Override
+    public void updateConceptById(ConceptEntry entry)
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/IndexService.java]: why is this method called ```updateConceptById``` if you give it a concept entry and not the id?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/IndexService.java [hunk 7]
```diff
@@ -120,4 +120,20 @@ public IndexingEvent getTotalNumberOfWordsIndexed() {
         return dao.getTotalNumberOfWordsIndexed();
     }
 
+    /**
+     * This method updates the concept in lucene index based on id of the
+     * concept
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/IndexService.java]: important thing here is that the concept is updated in lucene, not that its id is used.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptDeleteController.java [hunk 8]
```diff
@@ -155,7 +157,7 @@ public ModelAndView deleteConcept(@PathVariable("id") String id,
 			model.setViewName("welcome");
 			return model;
 		}
-		conceptManager.deleteConcept(id);
+        conceptManager.deleteConcept(id, principal.getName());
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptDeleteController.java]: indentation is off here and above

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/wrapper/ConceptEntryWrapper.java [hunk 9]
```diff
@@ -86,4 +88,20 @@ public String getUri() {
...
+        this.creatorId = creatorId;
+    }
+
+    public String getLastModifiedEvent() {
+        return lastModifiedEvent;
+    }
+
+    public void setLastModifiedEvent(String lastModifiedEvent) {
+        this.lastModifiedEvent = lastModifiedEvent;
+    }
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/wrapper/ConceptEntryWrapper.java]: indentation

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/xml/XMLConceptMessage.java [hunk 10]
```diff
@@ -82,12 +84,13 @@ public String getEntry(ConceptEntry entry, ConceptType type) {
...
-				.getCreatorId().trim() : ""));
-		sb.append("</" + XMLConstants.NAMESPACE_PREFIX + ":"
-				+ XMLConstants.CREATOR_ID + ">");
+        List<ChangeEvent> changeEvents = entry.getChangeEvents();
+        Collections.sort(changeEvents);
+
+        sb.append("<" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
+        sb.append(StringEscapeUtils.escapeXml10(
+                changeEvents.get(0).getUserName() != null ? changeEvents.get(0).getUserName().trim() : ""));
+        sb.append("</" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/xml/XMLConceptMessage.java]: indentation

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/root/DatabaseManager.java [hunk 11]
```diff
@@ -11,45 +11,47 @@
...
+
+    public void init() {
+        close();
+        ServerConfiguration configuration = Db4oClientServer.newServerConfiguration();
+        configuration.file().blockSize(80);
+        configuration.common().objectClass(ConceptEntry.class).objectField("wordnetId").indexed(true);
+        configuration.common().objectClass(ConceptEntry.class).objectField("id").indexed(true);
+        // Added to make sure list has been added to the database
+        // Ref: http://www.resolvinghere.com/sof/12343387.shtml
+        configuration.common().objectClass(ConceptEntry.class).updateDepth(2);
```
Comment (mkarthik90) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/root/DatabaseManager.java]: Added the above line

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/ConceptManager.java (hunk 4 above)
Comment (mkarthik90) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/ConceptManager.java]: Readded. Removed those while merging

**Comment by mkarthik90** [comment]
@jdamerow  Fixed the review comments

**Comment by jdamerow** [comment]
ok to test
